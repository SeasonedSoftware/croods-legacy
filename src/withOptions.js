import React from 'react'
import { Consumer } from './Context'

export default Component => props => (
  <Consumer>{options => <Component {...props} options={options} />}</Consumer>
)
