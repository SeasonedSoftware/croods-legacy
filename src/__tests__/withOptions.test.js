import React from 'react'
import renderer from 'react-test-renderer'

import withOptions from '../withOptions'

jest.mock('../Context', () => ({
  Consumer: ({ children }) => (
    <div>Consumer - {children({ foo: 'bar', quu: 'foo' })}</div>
  ),
}))

it('renders Component with props and options', () => {
  const props = { foo: 'quu', bar: 'baz' }
  const Component = withOptions(props => <div {...props}>Foo</div>)
  const tree = renderer.create(<Component {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
