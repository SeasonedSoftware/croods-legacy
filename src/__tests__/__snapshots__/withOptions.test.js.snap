// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Component with props and options 1`] = `
<div>
  Consumer - 
  <div
    bar="baz"
    foo="quu"
    quu="foo"
  >
    Foo
  </div>
</div>
`;
